include/communicator_quda.h

```diff
@@ -544,22 +514,48 @@ struct Communicator {
     Topology *topo = comm_create_topology(ndim, dims, rank_from_coords, map_data, comm_rank());
     comm_set_default_topology(topo);
 
+    
     // determine which GPU this rank will use
     char *hostname_recv_buf = (char *)safe_malloc(128 * comm_size());
     comm_gather_hostname(hostname_recv_buf);
 
-    // We only want one (1) gpuid for all communicators, so gpuid is static.
-    // We initialize gpuid if it's still negative.
-    if (gpuid < 0) {
+    if( gpuid < 0 ) { 
+
+      // Try PMI
+      char *local_rankstr = getenv( "PMI_RANK" );
+
+      // Try MVapich
+      if ( ! local_rankstr ) {
+        local_rankstr = getenv( "MV2_COMM_WORLD_LOCAL_RANK"  );
+      }
+
+      // Try OpenMPI
+      if ( ! local_rankstr ) {
+        local_rankstr = getenv( "OMPI_COMM_WORLD_LOCAL_RANK" );
+      }
 
-      gpuid = 0;
-      for (int i = 0; i < comm_rank(); i++) {
-        if (!strncmp(comm_hostname(), &hostname_recv_buf[128 * i], 128)) { gpuid++; }
+      // Try SLURM
+      if ( ! local_rankstr ) {
+       local_rankstr = getenv( "SLURM_LOCALID" );
       }
```

Comment:
So instead of trying to go over all possible MPI implementations, should we just query for, say `QUDA_COMM_WORLD_LOCAL_RANK`, and the user is responsible for setting it to the correct value?

Reply:
I think that's a much better way of doing things.  @bjoo FWIW since you wrote this.

I'll update this when I'm hacking on this again (next week), and also propagate this to my open hotfix PR for 1.1.

Reply:
I've merged in the release/1.1.x branch which reverts these additions: it was buggy anyway with respect to peer-to-peer setup and it wasn't needed anyway.